### PR TITLE
Change the mock value of port in TrinoBaseProfileMapping to an integer

### PR DESCRIPTION
**Pull request by dwolfeu**
## Description

The mock value of `port` in `TrinoBaseProfileMapping` has been changed to an integer.

## Related Issue(s)

Closes: #1315

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by dwolfeu** [comment]
I tested this using the [local docker-compose setup](https://astronomer.github.io/astronomer-cosmos/contributing#using-docker-compose-for-local-development), adding 

```
apache-airflow-providers-trino
dbt-trino
```

to `dev/requirements.txt`, and the dag I included to the [bug report](https://github.com/astronomer/astronomer-cosmos/issues/1315).

**Review by pankajkoti — COMMENTED** [review]
LGTM. Would it be possible to add a test for this that demonstrates how it would fail without this mock_profile method implementation and that with having it succeeds?

File: cosmos/profiles/trino/base.py [hunk 1]
```diff
@@ -41,6 +41,12 @@ def profile(self) -> dict[str, Any]:
         # remove any null values
         return self.filter_null(profile_vars)
 
+    @property
+    def mock_profile(self) -> dict[str, Any]:
```
Comment (tatiana) [inline cosmos/profiles/trino/base.py:45]: @dwolfeu, thank you so much for fixing this so quickly!

We're very close to merging this change. As @pankajkoti mentioned, adding a unit test to cover this change would ensure that we don't have a regression in this issue in the future.

You could base on these lines and do a similar check for `mock_profile.`
https://github.com/astronomer/astronomer-cosmos/blob/8ec46d2f680d6d5d4e71402030873c56d001adbc/tests/profiles/trino/test_trino_base.py#L37-L45

Please let us know if you have any questions - we are here to support you to get this to completion.

Comment (dwolfeu) [reply cosmos/profiles/trino/base.py:45]: I have added a test. It only checks whether the `mock_profile` is set correctly though; it does not do as @pankajkoti suggested, namely testing that `dbt ls` fails without the change and passes with it. I think I would need some help in writing such a test.

Just fyi: I'm on macOS (so Darwin) and I had to change `25beeb54cc4eeabe6198248e286a1cfe` in line 1582 to `6f63493009733a7be34364a6ea3ffd3c` (= the value in line 1584) in my local repo in order to get the unit test to pass.

https://github.com/astronomer/astronomer-cosmos/blob/8ec46d2f680d6d5d4e71402030873c56d001adbc/tests/dbt/test_graph.py#L1581-L1584

Comment (tatiana) [reply cosmos/profiles/trino/base.py:45]: Thanks for adding the unit test and reporting this issue with the test. I believe, for now, the unit test would already avoid us to have a regression on this - so I'm happy for us to merge this PR.

I logged a separate ticket related to the `hash_dir` test:
https://github.com/astronomer/astronomer-cosmos/issues/1329

**Review by tatiana — APPROVED** [review]

**Comment by pankajkoti** [comment]
Thanks a lot @dwolfeu for this great contribution & being so quick at contributing it as well as addressing feedback!

**Comment by dwolfeu** [comment]
No worries. It was a pleasure. Thank you @pankajkoti and @tatiana for your help!

**Pull request merged**